Remove hdf4 support

biagas opened the pull request:
### Description
Remove references to HDF4 from config-site files.
Remove Find module and build_visit module for HDF 4.
Delete HDF4 dependent readers ZeusMP, Rage, Cosmos.
Remove HDF4 support from Enzo reader.
Since Enzo is declared to be HDF5 dependent and won't be built without it, remove HDF5 conditional logic.
Remove HDF4 from masonry. However, didn't touch visit-version specific masonry files.

Resolves #18059

### Type of change

<!-- Please check one of the boxes below -->

* [ ] Bug fix~~
* [ ] New feature~~
* [ ] Documentation update~~
* [X] Other~~ <!-- please explain with a note below -->

### How Has This Been Tested?

There aren't any Enzo tests, but we have test data, so I ensured that VisIt can still read and create the same plots as prior to the change with that data.

### Checklist:

<!-- For items in this checklist that do not apply, simply insert two tilde chars, `~~`, just ahead of the left bracket char, `[` at the beginning of a line. Each line ends with two tilde chars to make doing such ~~strikeouts~~ easy. -->

- [X] I have followed the [style guidelines][1] of this project.~~
- [X] I have performed a self-review of my own code.~~
- [X] I have commented my code where applicable.~~
- [X] I have updated the release notes.~~
~~- [ ] I have made corresponding changes to the documentation.~~
~~- [ ] I have added debugging support to my changes.~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works.~~
~~- [ ] I have confirmed new and existing unit tests pass locally with my changes.~~
~~- [ ] I have added new baselines for any new tests to the repo.~~
~~- [ ] I have NOT made any changes to [*protocol* or *public interfaces*][3] in an RC branch.~~
- [X] I have assigned reviewers (see [VisIt's PR procedures][2] for more information).~~

[1]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/StyleGuide.html
[2]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers
[3]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/RCDevelopment.html#communication-protocols-and-public-apis

biagas committed:
Remove HDF4 support.

Remove references to HDF4 from config-site files.
Remove Find module and build_visit module for HDF 4.
Delete HDF4 dependent readers ZeusMP, Rage, Cosmos.
Remove HDF4 support from Enzo reader.
Since Enzo is declared to be HDF5 dependent and won't be built without it, remove HDF5 conditional logic.
Remove HDF4 from masonry. However, didn't touch visit-version specific masonry files.

biagas committed:
Merge branch 'develop' into task/biagas/remove_hdf4_support

markcmiller86 commented on src/resources/help/en_US/relnotes3.4.0.html:
```diff
@@ -54,7 +54,8 @@
 <ul>
-    <li>feature1</li>
+    <li>HDF 4 based readers were removed: Cosmos, RAGE, ZeusMP</li>
+    <li>HDF 4 support in the Enzo reader was removed</li>
```
Is it worth adding an item to `Build_changes` that HDF4 support was removed from VisIt?

biagas replied:
There is no 'Build_changes', there is 'Build_features' and the description is for features **_added_**.
There is also 'Configuration_changes', and 'Changes_for_VisIt_developers', which section would you suggest?

markcmiller86 replied:
`Build_features` and adjust the description to be something a little more general than just *added* features because I think that is its intent anyways. That said, another option is to just adjust the above slightly...

```suggestion
    <li>HDF4 support has been entirely removed from VisIt. This means Cosmos, RAGE and ZeusMP readers were removed and Enzo reader now supports only HDF5.</li>
```

brugger1 replied:
I would put it under "Build_features" and change the description in the section from "Build features added in version 3.4" to "Changes to build_visit in version 3.4". That would make it consistent with the description in the sections list at the top of the release notes. I would also change the description in the sections list from "Changes for build_visit" to "Changes to build_visit".

I also noticed that the sections list has "General_Features", "Advanced_Features" and "GUI_Changes". The second word in all those names should start with a lower case letter to be consistent with the rest of the names and to match the names further down.

biagas replied:
Updated.

markcmiller86 commented on src/tools/dev/masonry/opts/mb-trunk-darwin-10.11-x86_64.json:
```diff
@@ -26,4 +25,4 @@
                  "zlib"]}
-}}
\ No newline at end of file
+}}
```
I am inclined to leave these older config files unchnaged for posterity's sake. So, maybe not change this one or the `10.6` one or `linux-x86_64`.

biagas replied:
which linux-x86_64 file?

markcmiller86 replied:
[this one](https://github.com/visit-dav/visit/blob/468df249950bf5e85b9f7bd9db3442f950bf8594/src/tools/dev/masonry/opts/mb-trunk-linux-x86_64-release.json)

markcmiller86 reviewed (dismissed):
Nice! Feels great to get rid of stuff we no longer need 💪🏻.

biagas committed:
Revert changes to mb-trunk-darwin-10.11-x86_64.json per review request.

biagas committed:
Update release notes per review request.

Also updated the template file for major release to correspond to the request, so future versions would also have the corrections.

biagas committed:
Merge branch 'develop' into task/biagas/remove_hdf4_support
